react/features/toolbox/components/web/LeaveConferenceButton.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import { createToolbarEvent, sendAnalytics } from '../../../analytics';
+import { appNavigate } from '../../../app/actions';
+import { disconnect } from '../../../base/connection';
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import { AbstractButton, type AbstractButtonProps } from '../../../base/toolbox/components';
+
+/**
+ * The type of the React {@code Component} props of {@link LeaveConferenceButton}.
+ */
+type Props = AbstractButtonProps & {
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function
+};
+
+/**
+ * An implementation of an {@link AbstractButton} for leaving the meeting.
+ */
+class LeaveConferenceButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.leaveConference';
+    label = 'toolbar.leaveConference';
+    tooltip = 'toolbar.leaveConference';
+
+    /**
+     * Handles clicking / pressing the button.
+     *
+     * @override
+     * @protected
+     * @returns {void}
+     */
+    _handleClick() {
+        const { handleClick } = this.props;
+
+        if (handleClick) {
+            handleClick();
+
+            return;
+        }
+
+        sendAnalytics(createToolbarEvent('hangup'));
+
+        // FIXME: these should be unified.
+        if (navigator.product === 'ReactNative') {
```

Comment:
Since we now have this code in a couple of places how about we create a new action `leaveConference` which encapsulates the behavior?

Reply:
Component has been redone. This no longer applies.